runtime-sdk/src/types/token.rs

```diff
@@ -18,14 +16,18 @@ impl Denomination {
     /// Denomination in native token.
     pub const NATIVE: Denomination = Denomination(Vec::new());
 
-    // TODO: Enforce maximum length during deserialization.
```

Comment:
So you add `TryFrom` below, but the serialization I meant here was also CBOR serialization (currently provided by the derived `Deserialize` trait). Do we need a custom implementation that forwards to `Denomination::try_from`?